homeassistant/components/elmax/common.py

```diff
@@ -0,0 +1,229 @@
+                        self._panel_id,
+                    )
+                    raise ConfigEntryAuthFailed()
+
+                panel = panels[0]
+                self._panel_entry = panel
+
+                # If the panel is online, proceed with fetching its state
+                # and return it right away
+                if panel.online:
+                    status = await self._client.get_panel_status(
+                        control_panel_id=panel.hash, pin=self._panel_pin
+                    )  # type: PanelStatus
+
+                    # Store a dictionary for fast endpoint state access
+                    self._state_by_endpoint = {
+                        k.endpoint_id: k for k in status.all_endpoints
+                    }
+                    return status
+
+                # Otherwise, return None. Listeners will know that this means the device is offline
+                return None
+
+        except ElmaxBadPinError as err:
+            _LOGGER.error("Control panel pin was refused")
+            raise ConfigEntryAuthFailed from err
+        except ElmaxBadLoginError as err:
+            _LOGGER.error("Refused username/password")
+            raise ConfigEntryAuthFailed from err
+        except ElmaxApiError as err:
+            raise HomeAssistantError(
+                f"Error communicating with ELMAX API: {err}"
+            ) from err
+        except ElmaxNetworkError as err:
+            raise HomeAssistantError(
+                "Network error occurred while contacting ELMAX cloud"
+            ) from err
+        except Exception as err:
+            _LOGGER.exception("Unexpected exception")
+            raise HomeAssistantError("An unexpected error occurred") from err
+
+
+class ElmaxEntity(Entity):
+    """Wrapper for Elmax entities."""
+
+    def __init__(
+        self,
+        panel: PanelEntry,
+        elmax_device: DeviceEndpoint,
+        panel_version: str,
+        coordinator: ElmaxCoordinator,
+    ) -> None:
+        """Construct the object."""
+        self._panel = panel
+        self._device = elmax_device
+        self._panel_version = panel_version
+        self._coordinator = coordinator
+        self._transitory_state = None
+
+    @property
+    def transitory_state(self) -> Any | None:
+        """Return the transitory state for this entity."""
+        return self._transitory_state
+
+    @transitory_state.setter
+    def transitory_state(self, value: Any) -> None:
+        """Set the transitory state value."""
+        self._transitory_state = value
+
+    @property
+    def panel_id(self) -> str:
+        """Retrieve the panel id."""
+        return self._panel.hash
+
+    @property
+    def unique_id(self) -> str | None:
+        """Provide a unique id for this entity."""
+        return self._device.endpoint_id
+
+    @property
+    def name(self) -> str | None:
+        """Return the entity name."""
+        return self._device.name
```

Comment:
You can move this to `self._attr_name`  and `self._attr_unique_id` in `__init__`

Reply:
Users will be able to change the device name anyway in the UI since you're integration supports unique_ids

Reply:
Of course they would. What I am trying to say in here is that an user can change the device name from the Elmax external app. With the current implementation, if she does so, the change will reflect to HA, as the coordinator will update the device name and the property getter will return it accordingly. If we carve in stone the device name at the boot, any change occurred externally to HA (i.e. device name change from Elmax administration website) won't be reflected on HA. I believe this is something that should instead happen.

Reply:
And what I mean is that the name is not updated in the frontend if the user has changed the name in HA.

`Entity._async_write_ha_state`:
```python
        entry = self.registry_entry
        ...
        if (name := (entry and entry.name) or self.name) is not None:
            attr[ATTR_FRIENDLY_NAME] = name
```

IMHO You can leave it there, but I don't think it will work the way you describe it.